viadot/tasks/sap_bw.py

```diff
@@ -0,0 +1,122 @@
+import pandas as pd
+
+
+from typing import List, Dict, Any, Literal
+from prefect import Task
+from prefect.tasks.secrets import PrefectSecret
+from prefect.utilities import logging
+from viadot.exceptions import ValidationError
+from viadot.sources import SAPBW
+from viadot.task_utils import *
+
+logger = logging.get_logger()
+
+
+class SAPBWToDF(Task):
+    def __init__(
+        self,
+        sapbw_credentials: dict = None,
+        sapbw_credentials_key: str = "SAP",
+        env: str = "BW",
+        *args,
+        **kwargs,
+    ):
+        """
+        A task for quering the SAP BW (SAP Business Warehouse) source using pyrfc library.
+
+        Args:
+            sapbw_credentials (dict, optional): Credentials to SAP BW server. Defaults to None.
+            sapbw_credentials_key (str, optional): Azure KV secret. Defaults to "SAP".
+            env (str, optional): SAP environment. Defaults to "BW".
+        """
+        if sapbw_credentials is None:
+            self.sapbw_credentials = credentials_loader.run(
+                credentials_secret=sapbw_credentials_key
+            )
+            self.sapbw_credentials = self.sapbw_credentials[env]
+
+        else:
+            self.sapbw_credentials = sapbw_credentials
+
+        super().__init__(
+            name="sapbw_to_df",
+            *args,
+            **kwargs,
+        )
+
+    def __call__(self):
+        """Download SAP BW data to a DF"""
+        super().__call__(self)
+
+    def apply_user_mapping(
+        self, df: pd.DataFrame, mapping_dict: dict = {}
+    ) -> pd.DataFrame:
+        """
+        Function to apply the column mapping defined by user for the output dataframe.
+        DataFrame will be cut to selected columns - if any other columns need to be included in the output file,
+        please add them to the mapping dictionary with original names.
+
+        Args:
+            df (pd.DataFrame): Input dataframe for the column mapping task.
+            mapping_dict (dict, optional): Dictionary with original and new column names. Defaults to {}.
+
+        Returns:
+            pd.DataFrame: Output DataFrame with mapped columns.
+        """
+        self.logger.info("Applying user defined mapping for columns...")
+        df = df[mapping_dict.keys()]
+        df.columns = mapping_dict.values()
+
+        self.logger.info(f"Successfully applied user mapping.")
+
+        return df
+
+    def to_df(self, query_output: dict = {}) -> pd.DataFrame:
+        """
+        Function to convert the SAP output in JSON format into a dataframe.
+
+        Args:
+            query_output: SAP output dictionary in JSON format that contains data rows and column headers. Defaults to {}.
+
+        Returns:
+            pd.DataFrame: Output dataframe.
```

Comment:
Add the `Raises:` section to docstring as you are raising `ValidationError`.

Reply:
added
`        Raises:
            ValidationError: Prints the original SAP error message in case of issues with MDX execution.`